[Feature] 방장 설정 화면 UI 구현

Choyoonyoung98 opened the pull request:
## 개요
방장 설정 화면 UI 구현
<img width="200" alt="" src="https://user-images.githubusercontent.com/39290117/134366749-9b7a2e05-3283-4700-82e4-386c7f5a7248.PNG"><img width="200" alt="" src="https://user-images.githubusercontent.com/39290117/134366811-65908e5b-0bd7-4b9f-9122-b8cf7cd6eb72.PNG"><img width="200" alt="" src="https://user-images.githubusercontent.com/39290117/134366846-f41a0f39-5e13-4bc3-9ef6-c4fcd8738de4.PNG">

## 작업 내용
- [x] 방장 설정 메인 화면
- [x] 작성자 순서 변경
- [x] 작성 주기 변경

## 참고 사항
기존에 구현되어 있던 작성 주기 변경 팝업의 component 요소가 상화좌우에 border가 적용되어 있어서, 하단 border만 적용되도록 변경하려고 했으나 아직 구현하지 못했습니다. 이후 다시 구현하겠습니다! :)

related to #50

Choyoonyoung98 committed:
[CREATE] 필요 이미지 리소스 및 색상값 추가

Choyoonyoung98 committed:
[CREATE] 방장 설정 화면 UI 구현

Choyoonyoung98 committed:
[UPDATE] 이전 코멘트 반영

Choyoonyoung98 committed:
[UPDATE] 작성 순서 변경 UI 구현

- WritingOrderModifyCell: 작성순서 변경 collectionView cell 클래스 구현
- 작성자간 순서 변경을 위해 drag & drop 구현

Choyoonyoung98 committed:
[UPDATE] 비밀번호 변경 UI 화면 구현

- 키보드 이동에 따른 뷰 이동
- 텍스트필드 외부 뷰 선택 시, 키보드 내려가기 구현

Choyoonyoung98 committed:
[UPDATE] 작성 기간 변경 팝업 구체화 시도

- 이슈: 하단에만 border값을 주고싶은데 적용이 되지 않고 있습니다. 이후 다시 시도해보겠습니다.

jeon-soyeong commented on VODA_iOS/VODA_iOS/Sources/Diary/ViewController/Admin/AdminViewController.swift:
```diff
@@ -0,0 +1,49 @@
+class AdminViewController: UIViewController {
+    @IBOutlet weak var headerView: UIView!
+    private let writePeriodPopUpViewcontrollerIdentifier = "WritePeriodPopUpViewController"
+    private let updateInvitationCodeViewControllerIdentifier = "UpdateInvitationCodeViewController"
```
함수랑 한줄 띄어주면 좋을 것 같아요 :)

jeon-soyeong commented on VODA_iOS/VODA_iOS/Sources/Diary/ViewController/Admin/AdminViewController.swift:
```diff
@@ -0,0 +1,49 @@
+    private let updateInvitationCodeViewControllerIdentifier = "UpdateInvitationCodeViewController"
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Do any additional setup after loading the view.
```
안쓰는 주석 지우면 좋을 것 같아요!

jeon-soyeong commented on VODA_iOS/VODA_iOS/Sources/Diary/ViewController/Admin/AdminViewController.swift:
```diff
@@ -0,0 +1,49 @@
+        setStyle()
+    }
+    
+    func setStyle() {
```
private 넣으면 좋을 것 같아요!

jeon-soyeong commented on VODA_iOS/VODA_iOS/Sources/Diary/ViewController/Admin/UpdateInvitationCodeViewController.swift:
```diff
@@ -0,0 +1,74 @@
+    @IBOutlet weak var completeButtonView: UIView!
+    @IBOutlet weak var completeButton: UIButton!
+    
+    var originConstant: CGFloat = 0
```
변수랑 setStyle(), keyboardWillAppear(), keyboardWillDisappear() 해당 파일에서만 사용한다면 private 명시해주면 좋을 것 같아요!

jeon-soyeong commented on VODA_iOS/VODA_iOS/Sources/Diary/ViewController/Admin/WritingOrderModifyViewController+CollectionViewExtension.swift:
```diff
@@ -0,0 +1,99 @@
+    }
+}
+
+extension WritingOrderModifyViewController: UICollectionViewDragDelegate {
+    func collectionView(_ collectionView: UICollectionView, itemsForBeginning session: UIDragSession, at indexPath: IndexPath) -> [UIDragItem] {
+        let item = participants[indexPath.row]
+        let itemProvider = NSItemProvider(object: item as NSString)
+        let dragItem = UIDragItem(itemProvider: itemProvider)
+        dragItem.localObject = item
+        
+        return [dragItem]
+    }
+}
+
+extension WritingOrderModifyViewController: UICollectionViewDropDelegate {
+    func collectionView(_ collectionView: UICollectionView, dropSessionDidUpdate session: UIDropSession, withDestinationIndexPath destinationIndexPath: IndexPath?) -> UICollectionViewDropProposal {
+        if collectionView.hasActiveDrag {
+            return UICollectionViewDropProposal(operation: .move, intent: .insertAtDestinationIndexPath)
+        }
```
윤영님 시뮬 돌려보니까 드래그 앤 드롭 동작이 안되는 것 같아요..!!

Choyoonyoung98 replied:
이거는 개인적으로 말씀드렸지만.. 제 기기와 시뮬에서 재현이 안되는 상황이라 다시 확인해보겠습니다 ㅠㅜ

Choyoonyoung98 replied:
원인은 모르겠지만,, iOS 15 이전 OS는 
```
self.writingOrderCollectionView.dragInteractionEnabled = true
```
를 추가했더니 동작되는 것 같습니다

jeon-soyeong commented on VODA_iOS/VODA_iOS/Sources/Diary/ViewController/Admin/WritingOrderModifyViewController+CollectionViewExtension.swift:
```diff
@@ -0,0 +1,99 @@
+        }
+    }
+    
+    fileprivate func reorderItems(coordinator: UICollectionViewDropCoordinator, destinationIndexPath: IndexPath, collectionView: UICollectionView) {
```
해당 함수만 fileprivate를 붙이는 이유가 따로 있는지 궁금해용!

Choyoonyoung98 replied:
그냥 단순히 다른 소스 파일에서 접근하지 않아도 되는 함수이기 때문에 `fileprivate` 라고 붙였습니다!

위에서 말씀해주신 것처럼 다른 함수들에 대해서도 접근 제어 관리 확인해보겠습니당

jeon-soyeong replied:
위 함수들도 다른 소스 파일에서 접근하지 않아도 되는데 요기만 따로 붙이셔서 혹시 다른 이유가 있나 했어용! 
이유가 딱히 없다면 혹시 통일해서 쓰는 것은 어떨까요~?

jeon-soyeong commented on VODA_iOS/VODA_iOS/Sources/Diary/ViewController/Admin/WritingOrderModifyViewController.swift:
```diff
@@ -0,0 +1,33 @@
+import UIKit
+
+class WritingOrderModifyViewController: UIViewController {
+    @IBOutlet weak var completButtonView: UIView!
```
completeButtonView 오타 수정해주시면 될 것 같아요 :)

jeon-soyeong commented on VODA_iOS/VODA_iOS/Sources/PopUp/ViewController/WritePeriodPopUpViewController.swift:
```diff
@@ -23,6 +23,10 @@ class WritePeriodPopUpViewController: UIViewController {
+    
     @IBAction func selectWritePeriod(_ sender: Any) {
         //TODO: 선택된 작성 주기 전달하기
         //        periodList[selectedRow]
```
여기서 그전 화면으로 데이터 보내는 처리도 해주시면 될 것 같아요! :)

Choyoonyoung98 replied:
데이터 처리는 네트워킹 때 하려고 밍기적 미뤄뒀었는데 이번 PR에 추가 구현하겠습니다 :)

jeon-soyeong reviewed (commented):
시뮬 돌려보니까 참여 코드 재설정화면 완료 버튼에 위에 'button' 글씨가 중복으로 적혀 있어요 삭제해주시면 될 것 같아요 :)
(+ 디자인 그리고 여기 비밀번호 변경도 한뎁스 더 들어가니까 화살표가 추가되고, 비밀번호 or 참여코드로 워딩이 통일 되면 좋을 것 같은데 안나님께 말씀드려봐도 될까요??)

작성 순서 변경 화면은 가운데 정렬 안되어 있는 거랑 선 길이, 완료버튼 길이, 옆에 세줄 버튼이 없는 것, 그리고 드래그앤 드롭이 안되는 것 같은데 확인부탁드립니다!

Choyoonyoung98 committed:
[UPDATE] 코멘트 내용 반영

- drag & drop iOS 14 이하에서 동작되지 않는 현상 해결
- 변수명 오타 수정
- delegate 패턴을 통해 관리자 홈 페이지로 작성 주기 데이터 전달
- 필요없는 주석 제거

Choyoonyoung98 commented:
수정했습니다 drag & drop 부분 확인부탁드려요!

jeon-soyeong commented on VODA_iOS/VODA_iOS/Sources/Diary/ViewController/Admin/AdminViewController.swift:
```diff
@@ -9,12 +9,14 @@ import UIKit
+    
     override func viewDidLoad() {
         super.viewDidLoad()
 
```
viewDidLoad 내부에 추가될 코드 없으면 viewDidLoad 함수 내부 공백 없으면 더 좋을 것 같아요~!

jeon-soyeong commented on VODA_iOS/VODA_iOS/Sources/Diary/ViewController/Admin/AdminViewController.swift:
```diff
@@ -41,9 +43,16 @@ class AdminViewController: UIViewController {
-        dvc.modalPresentationStyle = .overCurrentContext
-        dvc.modalTransitionStyle = .crossDissolve
-        self.present(dvc, animated: true, completion: nil)
+        let dvc = UIStoryboard(name: "PopUp", bundle: nil).instantiateViewController(withIdentifier: writePeriodPopUpViewcontrollerIdentifier) as? WritePeriodPopUpViewController
+        dvc?.modalPresentationStyle = .overCurrentContext
+        dvc?.modalTransitionStyle = .crossDissolve
+        dvc?.writePeriodDelegate = self
+        self.present(dvc ?? UIViewController(), animated: true, completion: nil)
```
여기 dvc 약어를 풀어서 쓰거나 writePeriodPopUpViewController 적으면 좋을 것 같습니다!

jeon-soyeong commented on VODA_iOS/VODA_iOS/Sources/Diary/ViewController/Admin/WritingOrderModifyViewController+CollectionViewExtension.swift:
```diff
@@ -46,37 +41,40 @@ extension WritingOrderModifyViewController: UICollectionViewDragDelegate {
-           let sourceIndexPath = item.sourceIndexPath {
-            collectionView.performBatchUpdates({
-                participants.remove(at: sourceIndexPath.item)
+    fileprivate func reorderItems(coordinator: UICollectionViewDropCoordinator, sourceIndexPath: IndexPath, destinationIndexPath: IndexPath, dropItem: UICollectionViewDropItem, collectionView: UICollectionView) {
+        collectionView.performBatchUpdates({
+            participants.remove(at: sourceIndexPath.item)
```
위 함수들도 다른 소스 파일에서 접근하지 않아도 되는데 요기만 따로 붙이셔서 혹시 다른 이유가 있나 했어용!
이유가 딱히 없다면 혹시 통일해서 쓰는 것은 어떨까요~?

jeon-soyeong commented on VODA_iOS/VODA_iOS/Sources/Diary/ViewController/Admin/WritingOrderModifyViewController.swift:
```diff
@@ -28,6 +30,6 @@ class WritingOrderModifyViewController: UIViewController {
         self.setNavigationBarColor(color: UIColor.clear)
         self.setBackButton(color: .black)
         
-        self.completButtonView.makeCornerRadius(radius: 16)
```
해당 파일에서 setStyle로 쓰여 있는데 여기도 private 붙이면 좋을 것 같아요 :)

jeon-soyeong reviewed (commented):
윤영님 드래그앤드롭 코드 한줄 추가해주셔서 되는 것 확인했습니다~
현재 두 화면에서 누락된 수정 사항이 있는 것 같아요 확인 부탁드려요~ :)

[비밀번호 변경 화면] 사진 원형 적용, 셀 밑에 선 길이, 완료 버튼 길이 수정 누락, 완료 버튼 dismiss 동작 안됨
[작성순서 변경 화면] 완료 버튼 dismiss 동작 안됨

Choyoonyoung98 committed:
[UPDATE] 코멘트 내용 반영

Choyoonyoung98 commented:
코멘트 내용 반영했습니다! 확인 부탁드려요~
